code/modules/client/preferences_ui.dm

```diff
@@ -588,48 +570,49 @@
 			else if(!current_client.tooltips && tooltips)
 				current_client.tooltips = new /datum/tooltip(current_client)
 
-		if("keybindings_set")
-			var/kb_name = params["keybinding"]
+		if("set_keybind")
+			var/kb_name = params["keybind_name"]
 			if(!kb_name)
 				return
 
-			var/clear_key = text2num(params["clear_key"])
 			var/old_key = params["old_key"]
-			if(clear_key)
-				if(key_bindings[old_key])
-					key_bindings[old_key] -= kb_name
-					if(!length(key_bindings[old_key]))
-						key_bindings -= old_key
+			if(key_bindings[old_key])
+				key_bindings[old_key] -= kb_name
+				if(!length(key_bindings[old_key]))
+					key_bindings -= old_key
+			
+			if(!params["key"])
+				return
+			var/mods = params["key_mods"]
+			var/full_key = params["key"]
+			var/AltMod = ("ALT" in mods) ? "Alt" : ""
+			var/CtrlMod = ("CONTROL" in mods) ? "Ctrl" : ""
+			var/ShiftMod = ("SHIFT" in mods) ? "Shift" : ""
+			full_key = AltMod + CtrlMod + ShiftMod + full_key
```

Comment:
Firstly camelcase
Secondly does this even work

Reply:
This works BECAUSE of camel case.

Tgui has them in full caps, but the keybind system has them like that

Reply:
Yes, this works. Tgui has the key in full caps, and byond has them with one cap letter at the beginning